### ISSUE TITLE
Rework SCA Policy for Apple macOS 13.0 Ventura

### ISSUE THREAD

**Issue by jk-olaoluwa** [wazuh/wazuh#17319]
| Component | Action type           | Main Issue 
| --- | --- | ---
| SCA       | Create |  #18306

## Main tasks

- [x] Use the latest CIS benchmark PDF Draft version (We are updating to version 1.1.0, we will update to version 2.0.0 soon)
- [x] Verify IDs numbers.
- [x] Verify texts are correct: Title, Description, Rationale and Remediation.
- [x] Verify Compliance: CIS, CIS_CSC.
- [x] Verify condition and rules:
  - [x] To Pass.
  - [x] To Fail.

### Syntax and semantic

- [x] a) ID of each policy must be contiguous.
- [x] b) The order and format set in [Documentation](https://documentation.wazuh.com/current/user-manual/capabilities/sec-config-assessment/creating-custom-policies.html) must be respected.
- [x] c) YML must be valid to avoid errors.

### Content

- [x] a) Compare each check with its analog from CIS Benchmark.
- [x] b) Try maintaining each rule as similar as possible with the `Audit` section from the CIS check.
- [x] c) Check that the commands provide the expected output.
- [x] d) When a failure is discovered, check similar policies to avoid repetition of the issue.

### Unit testing

- [x] a) Output from `ossec.log` after the SCA scan and a raw output of the result of the checks.
<details><summary>Tests results</summary>
<p>

#### Analysisd (server or local)
analysisd.debug=2

#### Auth daemon debug (server)
authd.debug=0

#### Exec daemon debug (server, local, or Unix agent)
execd.debug=0

#### Monitor daemon debug (server, local, or Unix agent)
monitord.debug=0

#### Log collector (server, local or Unix agent)
logcollector.debug=0

#### Integrator daemon debug (server, local, or Unix agent)
integrator.debug=0

#### Unix agentd
agent.debug=2

</p>

```
2023/11/02 02:22:08 sca[24467] wm_sca.c:493 at wm_sca_read_files(): DEBUG: Calculating hash for scanned results.
2023/11/02 02:22:08 sca[24467] wm_sca.c:2817 at wm_sca_hash_integrity(): DEBUG: Concatenating check results:
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30000; Result: 'failed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30001; Result: 'passed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30002; Result: 'passed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30003; Result: 'passed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30004; Result: 'failed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30005; Result: 'passed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30006; Result: 'passed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30007; Result: 'failed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30008; Result: 'failed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30009; Result: 'passed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30010; Result: 'passed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30011; Result: 'passed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30012; Result: 'failed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30013; Result: 'passed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30014; Result: 'passed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30015; Result: 'passed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30016; Result: 'passed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30017; Result: 'passed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30018; Result: 'passed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30019; Result: 'passed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30020; Result: 'passed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30021; Result: 'passed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30022; Result: 'failed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30023; Result: 'failed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30024; Result: 'passed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30025; Result: 'failed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30026; Result: 'failed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30027; Result: 'failed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30028; Result: 'passed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30029; Result: 'passed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30030; Result: 'failed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30031; Result: 'failed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30032; Result: 'failed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30033; Result: 'passed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30034; Result: 'passed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30035; Result: 'passed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30036; Result: 'passed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30037; Result: 'passed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30038; Result: 'passed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30039; Result: 'failed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30040; Result: 'failed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30041; Result: 'failed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30042; Result: 'failed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30043; Result: 'passed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30044; Result: 'failed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30045; Result: 'passed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30046; Result: 'passed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30047; Result: 'passed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30048; Result: 'failed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30049; Result: 'failed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30050; Result: 'failed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30051; Result: 'failed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30052; Result: 'failed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30053; Result: 'failed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30054; Result: 'failed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30055; Result: 'failed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30056; Result: 'passed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30057; Result: 'passed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30058; Result: 'passed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30059; Result: 'failed'
2023/11/02 02:22:08 sca[24467] wm_sca.c:2820 at wm_sca_hash_integrity(): DEBUG: ID: 30060; Result: 'passed'
2023/11/02 02:22:11 sca[24467] wm_sca.c:2450 at wm_sca_send_summary(): DEBUG: Sending summary event for file: 'cis_macOS_13.0_Ventura.yml'
2023/11/02 02:22:11 sca[24467] wm_sca.c:270 at wm_sca_send_alert(): DEBUG: Sending event: {"type":"summary","scan_id":1672210278,"name":"CIS_Apple_macOS_13.0_Ventura_Benchmark_v1.1.0","policy_id":"cis_macOS_13.0_Ventura.yml","file":"cis_macOS_13.0_Ventura.yml","description":"This document provides prescriptive guidance for establishing a secure configuration posture for MacOS 13 Ventura systems.","references":"https://www.cisecurity.org/cis-benchmarks/","passed":34,"failed":27,"invalid":0,"total_checks":61,"score":55.737705230712891,"start_time":1698916913,"end_time":1698916928,"hash":"995d5e77d695bf417df4d60f7e34d19b881853a1d03b90a2fe7a9b643cdf4b68","hash_file":"3dec3105d06639923deae5b7db1118d429f093aeafb867a48700abd12a99f9e9","first_scan":1}
```

</details>

### Deployment

- [x] a) If the policy it's new, it must be added to the `sca.files` templates.
- [x] b) If the OS has many supported SCA policies, a policy must be set as the default policy. ([as example](https://github.com/wazuh/wazuh/blob/master/etc/templates/config/centos/sca.files))

### Documentation

- [x] a) Ensure documentation SCA list includes the created or updated SCA.

**Comment by Johnng007** [comment]
Section 1.1 - 2.1 - Completed

**Comment by Johnng007** [comment]
Section 21. - 3.1 - Complete
https://github.com/wazuh/wazuh/pull/19037/commits/3961a07dbf129a363eb275535bce53491aab676a

**Comment by jk-olaoluwa** [comment]
### Update 28/09/2023

#### Setting up environment for test

**Comment by ooniagbi** [comment]
The test was successful. The results have been added in the first comment above.

**Comment by IsExec** [comment]
# macOS 13.0 Ventura SCA Testing Report
**Condition**: The reworked SCA was benchmarked against macOS 13.0 old SCA.  

## Agent running on old SCA (macOS 13.0 SCA)

###  Endpoint spec

<details>
<summary>CPU</summary>

````console
# system_profiler SPHardwareDataType
Hardware:

    Hardware Overview:

      Model Name: Mac
      Model Identifier: Parallels19,1
      Processor Name: Unknown
      Processor Speed: 3.19 GHz
      Number of Processors: 1
      Total Number of Cores: 2
      L2 Cache (per Core): 256 KB
      L3 Cache: 12 MB
      Memory: 4 GB
      System Firmware Version: 19.3.0 (54924)
      OS Loader Version: 577.140.2~22
      SMC Version (system): 1.16f8
````

</details>

<details>
<summary>RAM</summary>

````console
# vm_stat
Mach Virtual Memory Statistics: (page size of 4096 bytes)
Pages free:                               31687.
Pages active:                            322169.
Pages inactive:                          394614.
Pages speculative:                       107457.
Pages throttled:                              0.
Pages wired down:                        175670.
Pages purgeable:                           2158.
"Translation faults":                  61588876.
Pages copy-on-write:                    4633041.
Pages zero filled:                     29875663.
Pages reactivated:                        15733.
Pages purged:                             38312.
File-backed pages:                       415905.
Anonymous pages:                         408335.
Pages stored in compressor:               38305.
Pages occupied by compressor:             16276.
Decompressions:                            5728.
Compressions:                             44668.
Pageins:                                 772187.
Pageouts:                                  1036.
````

</details>

<details>
<summary>Disk</summary>

````console
# df -h
Filesystem                                                   Size   Used  Avail Capacity iused      ifree %iused  Mounted on
/dev/disk1s5s1                                               95Gi   15Gi   68Gi    18%  356810  713313080    0%   /
devfs                                                       187Ki  187Ki    0Bi   100%     646          0  100%   /dev
/dev/disk1s2                                                 95Gi  3.5Gi   68Gi     5%     938  713313080    0%   /System/Volumes/Preboot
/dev/disk1s4                                                 95Gi  1.0Mi   68Gi     1%       1  713313080    0%   /System/Volumes/VM
/dev/disk1s6                                                 95Gi   62Mi   68Gi     1%     651  713313080    0%   /System/Volumes/Update
/dev/disk1s1                                                 95Gi  7.0Gi   68Gi    10%  191921  713313080    0%   /System/Volumes/Data
/dev/disk0s3                                                161Gi  288Mi  161Gi     1%     101 4294967178    0%   /Volumes/Untitled
/dev/disk1s5                                                 95Gi   15Gi   68Gi    18%  403757  713313080    0%   /System/Volumes/Update/mnt1
map auto_home                                                 0Bi    0Bi    0Bi   100%       0          0  100%   /System/Volumes/Data/home
//guest:@Shared%20Folders._smb._tcp.local/Shared%20Folders  4.0Ti    0Bi  4.0Ti     0% 18446744073709551614 1073741824 1717986921600%   /Volumes/SharedFolders
````
</details>


### Footprint test results
<details>
<summary>Metrics</summary>

![cpu_percent_plot](https://github.com/user-attachments/assets/977be411-ed6f-40a7-b01b-fa3385784b5e)
![disk_percent_plot](https://github.com/user-attachments/assets/aa9c5808-a9c9-477c-a70b-ae3825de58ed)
![disk_used_(mb)_plot](https://github.com/user-attachments/assets/c1b7c588-32ad-4b45-a1e6-926a579308b6)
![memory_percent_plot](https://github.com/user-attachments/assets/8f997a6d-34eb-4b0b-89d1-a401f8eea979)
![memory_used_(mb)_plot](https://github.com/user-attachments/assets/473209b8-e03c-4ef8-8aac-efc42d622f6b)
![ProcessesCPU](https://github.com/user-attachments/assets/bbfd1f98-8db6-41ce-b640-c0124a45cca6)
![ProcessesCPU_Change_Point](https://github.com/user-attachments/assets/cb2e3902-c97b-4fe7-848d-a506ce326997)
![ProcessesCPUlegend](https://github.com/user-attachments/assets/ce2b2562-1f72-49e2-b55a-40fef8fc7da0)
![ProcessesRAM](https://github.com/user-attachments/assets/0b040436-46d4-452f-b523-6a11b2c490ca)
![ProcessesRAM_Change_Point](https://github.com/user-attachments/assets/a5bcf3c7-b6ed-4ce6-aa36-e4a63fadb09d)
![ProcessesRAMlegend](https://github.com/user-attachments/assets/6dc7b0a7-7a1a-40a7-a69a-67ec8e190cea)
![swap_percent_plot](https://github.com/user-attachments/assets/176332df-9f37-4836-9c76-0529c5108137)
![swap_used_(mb)_plot](https://github.com/user-attachments/assets/550dd5d1-1fe3-4d75-bd6e-5c132d581477)
</details>

<details>
<summary>SCA logs</summary>

````console
# tail /Library/Ossec/logs/ossec.log
2024/08/01 19:58:40 sca: INFO: Evaluation finished for policy '/Library/Ossec/ruleset/sca/cis_apple_macOS_13.x.yml'
2024/08/01 19:58:40 sca: INFO: Security Configuration Assessment scan finished. Duration: 13 seconds.
2024/08/01 20:03:27 sca: INFO: Starting Security Configuration Assessment scan.
2024/08/01 20:03:27 sca: INFO: Starting evaluation of policy: '/Library/Ossec/ruleset/sca/cis_apple_macOS_13.x.yml'
2024/08/01 20:03:43 sca: INFO: Evaluation finished for policy '/Library/Ossec/ruleset/sca/cis_apple_macOS_13.x.yml'
2024/08/01 20:03:43 sca: INFO: Security Configuration Assessment scan finished. Duration: 16 seconds.
2024/08/01 20:08:27 sca: INFO: Starting Security Configuration Assessment scan.
2024/08/01 20:08:27 sca: INFO: Starting evaluation of policy: '/Library/Ossec/ruleset/sca/cis_apple_macOS_13.x.yml'
2024/08/01 20:08:42 sca: INFO: Evaluation finished for policy '/Library/Ossec/ruleset/sca/cis_apple_macOS_13.x.yml'
2024/08/01 20:08:42 sca: INFO: Security Configuration Assessment scan finished. Duration: 15 seconds.
````

</details>

[metrics.zip](https://github.com/user-attachments/files/16465385/metrics.zip)


## Agent running on new SCA (macOS 13)

### Endpoint spec
<details>
<summary>CPU</summary>

````console
# system_profiler SPHardwareDataType
Hardware:

    Hardware Overview:

      Model Name: Mac
      Model Identifier: Parallels19,1
      Processor Name: Unknown
      Processor Speed: 3.19 GHz
      Number of Processors: 1
      Total Number of Cores: 2
      L2 Cache (per Core): 256 KB
      L3 Cache: 12 MB
      Memory: 4 GB
      System Firmware Version: 19.3.0 (54924)
      OS Loader Version: 577.140.2~22
      SMC Version (system): 1.16f8
````

</details>

<details>
<summary>RAM</summary>

````console
# vm_stat
Mach Virtual Memory Statistics: (page size of 4096 bytes)
Pages free:                               28894.
Pages active:                            327682.
Pages inactive:                          438873.
Pages speculative:                        60248.
Pages throttled:                              0.
Pages wired down:                        179527.
Pages purgeable:                           2457.
"Translation faults":                  59836645.
Pages copy-on-write:                    4133820.
Pages zero filled:                     30581274.
Pages reactivated:                        13679.
Pages purged:                             35681.
File-backed pages:                       410327.
Anonymous pages:                         416476.
Pages stored in compressor:               31558.
Pages occupied by compressor:             12717.
Decompressions:                            6236.
Compressions:                             38296.
Pageins:                                 745391.
Pageouts:                                  1092.
````

</details>

<details>
<summary>Disk</summary>

````console
# df -h
Filesystem                                                   Size   Used  Avail Capacity iused      ifree %iused  Mounted on
/dev/disk1s5s1                                               95Gi   15Gi   68Gi    18%  356810  713445680    0%   /
devfs                                                       187Ki  187Ki    0Bi   100%     646          0  100%   /dev
/dev/disk1s2                                                 95Gi  3.5Gi   68Gi     5%     938  713445680    0%   /System/Volumes/Preboot
/dev/disk1s4                                                 95Gi  1.0Mi   68Gi     1%       1  713445680    0%   /System/Volumes/VM
/dev/disk1s6                                                 95Gi   62Mi   68Gi     1%     651  713445680    0%   /System/Volumes/Update
/dev/disk1s1                                                 95Gi  7.0Gi   68Gi    10%  192072  713445680    0%   /System/Volumes/Data
/dev/disk0s3                                                161Gi  288Mi  161Gi     1%     101 4294967178    0%   /Volumes/Untitled
/dev/disk1s5                                                 95Gi   15Gi   68Gi    18%  403757  713445680    0%   /System/Volumes/Update/mnt1
map auto_home                                                 0Bi    0Bi    0Bi   100%       0          0  100%   /System/Volumes/Data/home
//guest:@Shared%20Folders._smb._tcp.local/Shared%20Folders  4.0Ti    0Bi  4.0Ti     0% 18446744073709551614 1073741824 1717986921600%   /Volumes/SharedFolders
````
</details>


### Footprint test results
<details>
<summary>Metrics</summary>

![cpu_percent_plot](https://github.com/user-attachments/assets/e9b402ee-46d5-495d-9f43-99124f441d8a)
![disk_percent_plot](https://github.com/user-attachments/assets/a5aea306-6015-41e5-8f02-218b7377affe)
![disk_used_(mb)_plot](https://github.com/user-attachments/assets/9f60e8a9-f1e0-4b72-a254-e413f468553c)
![memory_percent_plot](https://github.com/user-attachments/assets/3ef300ba-e34e-40d6-ad16-73b350780d32)
![memory_used_(mb)_plot](https://github.com/user-attachments/assets/7f14c82c-13a0-457e-a989-32b65b40d679)
![ProcessesCPU](https://github.com/user-attachments/assets/ec239071-8e66-478f-9586-63f82e6f5c28)
![ProcessesCPU_Change_Point](https://github.com/user-attachments/assets/fb621c58-1df8-4d37-8fe1-3b94502f2bb0)
![ProcessesCPUlegend](https://github.com/user-attachments/assets/cda16879-a18e-4dd0-838a-de32e41fe1aa)
![ProcessesRAM](https://github.com/user-attachments/assets/b9b89df1-1c1b-4c48-937f-990ecfb610a4)
![ProcessesRAM_Change_Point](https://github.com/user-attachments/assets/3d9bbee0-2c9f-4de4-a523-685c895e1293)
![ProcessesRAMlegend](https://github.com/user-attachments/assets/b47a535f-9883-4a14-afac-11a136e73521)
![swap_percent_plot](https://github.com/user-attachments/assets/8a94ecd9-1313-4dca-9a61-0c969727e9da)
![swap_used_(mb)_plot](https://github.com/user-attachments/assets/c2a7571d-94a4-4963-830e-ab3df5371553)
</details>

<details>
<summary>SCA logs</summary>

````console
2024/08/01 19:50:20 sca: INFO: Evaluation finished for policy '/Library/Ossec/ruleset/sca/cis_apple_macOS_13.x.yml'
2024/08/01 19:50:20 sca: INFO: Security Configuration Assessment scan finished. Duration: 13 seconds.
2024/08/01 19:55:07 sca: INFO: Starting Security Configuration Assessment scan.
2024/08/01 19:55:07 sca: INFO: Starting evaluation of policy: '/Library/Ossec/ruleset/sca/cis_apple_macOS_13.x.yml'
2024/08/01 19:55:31 sca: INFO: Evaluation finished for policy '/Library/Ossec/ruleset/sca/cis_apple_macOS_13.x.yml'
2024/08/01 19:55:31 sca: INFO: Security Configuration Assessment scan finished. Duration: 24 seconds.
2024/08/01 20:00:07 sca: INFO: Starting Security Configuration Assessment scan.
2024/08/01 20:00:07 sca: INFO: Starting evaluation of policy: '/Library/Ossec/ruleset/sca/cis_apple_macOS_13.x.yml'
2024/08/01 20:00:29 sca: INFO: Evaluation finished for policy '/Library/Ossec/ruleset/sca/cis_apple_macOS_13.x.yml'
2024/08/01 20:00:29 sca: INFO: Security Configuration Assessment scan finished. Duration: 22 seconds.
````

</details>


[metrics.zip](https://github.com/user-attachments/files/16465440/metrics.zip)


## Analysis

  | Benchmark   (macOS 13 Old) | New SCA (macOS   13 rework) | Difference
-- | -- | -- | --
CPU |   |   |  
`wazuh-modulesd` average   CPU percent | 1.2 | 2.3 | 1.1
Overall   average CPU percent | 30.1 | 22.94 | -7.16
Overall peak   CPU percent | 59.9 | 56.4 | -3.5
Memory |   |   | 0
`wazuh   modulesd` average memory (MB) | 0 | 0 |  
Overall   average memory used (MB) | 1947.44 | 1971.93 | 24.49
Overall peak   memory used peak (MB) | 1969.49 | 1987.71 | 18.22
Disk |   |   |  
Overall   average disk used (MB) | 15152.34 | 15152.34 | 0
Overall peak   disk used (MB) | 15152.34 | 15152.34 | 0
Overall   average disk read bytes | 3479.16 | 3491.69 | 12.53
Overall peak   disk read bytes | 3479.26 | 3491.75 | 12.49
Overall   average disk write bytes | 6686.76 | 6651.95 | -34.81
Overall peak   disk write bytes | 6688.5 | 6652.5 | -36

**Comment by IsExec** [comment]
## Note on Spike in CPU


Overall peak CPU percent | 59.9 | 56.4
-- | -- | --

From the top CPU and debug logs, the softwareupdate process used by check 1.1 (Ensure All Apple-provided Software Is Current) is the suspected cause of the spike.

### Rule: - "c:softwareupdate -l -> r:No new software available."

**Comment by IsExec** [comment]
# Analysis based on New SCA without Check 1.1

## Agent running on new SCA (macOS 13)

### Endpoint spec
<details>
<summary>CPU</summary>

````console
# system_profiler SPHardwareDataType
Hardware:

    Hardware Overview:

      Model Name: Mac
      Model Identifier: Parallels19,1
      Processor Name: Unknown
      Processor Speed: 3.19 GHz
      Number of Processors: 1
      Total Number of Cores: 2
      L2 Cache (per Core): 256 KB
      L3 Cache: 12 MB
      Memory: 4 GB
      System Firmware Version: 19.3.0 (54924)
      OS Loader Version: 577.140.2~22
      SMC Version (system): 1.16f8
````

</details>

<details>
<summary>RAM</summary>

````console
# vm_stat
Mach Virtual Memory Statistics: (page size of 4096 bytes)
Pages free:                               28894.
Pages active:                            327682.
Pages inactive:                          438873.
Pages speculative:                        60248.
Pages throttled:                              0.
Pages wired down:                        179527.
Pages purgeable:                           2457.
"Translation faults":                  59836645.
Pages copy-on-write:                    4133820.
Pages zero filled:                     30581274.
Pages reactivated:                        13679.
Pages purged:                             35681.
File-backed pages:                       410327.
Anonymous pages:                         416476.
Pages stored in compressor:               31558.
Pages occupied by compressor:             12717.
Decompressions:                            6236.
Compressions:                             38296.
Pageins:                                 745391.
Pageouts:                                  1092.
````

</details>

<details>
<summary>Disk</summary>

````console
# df -h
Filesystem                                                   Size   Used  Avail Capacity iused      ifree %iused  Mounted on
/dev/disk1s5s1                                               95Gi   15Gi   68Gi    18%  356810  713445680    0%   /
devfs                                                       187Ki  187Ki    0Bi   100%     646          0  100%   /dev
/dev/disk1s2                                                 95Gi  3.5Gi   68Gi     5%     938  713445680    0%   /System/Volumes/Preboot
/dev/disk1s4                                                 95Gi  1.0Mi   68Gi     1%       1  713445680    0%   /System/Volumes/VM
/dev/disk1s6                                                 95Gi   62Mi   68Gi     1%     651  713445680    0%   /System/Volumes/Update
/dev/disk1s1                                                 95Gi  7.0Gi   68Gi    10%  192072  713445680    0%   /System/Volumes/Data
/dev/disk0s3                                                161Gi  288Mi  161Gi     1%     101 4294967178    0%   /Volumes/Untitled
/dev/disk1s5                                                 95Gi   15Gi   68Gi    18%  403757  713445680    0%   /System/Volumes/Update/mnt1
map auto_home                                                 0Bi    0Bi    0Bi   100%       0          0  100%   /System/Volumes/Data/home
//guest:@Shared%20Folders._smb._tcp.local/Shared%20Folders  4.0Ti    0Bi  4.0Ti     0% 18446744073709551614 1073741824 1717986921600%   /Volumes/SharedFolders
````
</details>


### Footprint test results
<details>
<summary>Metrics</summary>

![cpu_percent_plot](https://github.com/user-attachments/assets/699b55a3-6b3f-4703-af74-1ec5133d7f82)
![disk_percent_plot](https://github.com/user-attachments/assets/489fabc1-8527-490d-b6a5-89d1eb75d2ac)
![disk_used_(mb)_plot](https://github.com/user-attachments/assets/e7478fbb-90c0-42c8-8511-91f2fd4fe773)
![memory_percent_plot](https://github.com/user-attachments/assets/9b022d79-d772-41da-be6a-e990dab5d352)
![memory_used_(mb)_plot](https://github.com/user-attachments/assets/c1c4af59-dc6f-41cc-87bf-2aa990c83281)
![ProcessesCPU](https://github.com/user-attachments/assets/5bc7ec27-3514-42fa-93be-ec8e7c9ffedd)
![ProcessesCPU_Change_Point](https://github.com/user-attachments/assets/70415612-c5ed-4fe2-8284-68215449dba9)
![ProcessesCPUlegend](https://github.com/user-attachments/assets/cda16879-a18e-4dd0-838a-de32e41fe1aa)
![ProcessesRAM](https://github.com/user-attachments/assets/f2b7abf7-3312-497f-ad5f-a48fe5eb68b0)
![ProcessesRAM_Change_Point](https://github.com/user-attachments/assets/a1904124-1b64-4691-9c7c-fa48c35a574f)
![ProcessesRAMlegend](https://github.com/user-attachments/assets/ec93f201-43f3-4f87-8200-0d560d86a08b)
![swap_percent_plot](https://github.com/user-attachments/assets/b1d43acb-feb3-406e-8265-e1c7ce8ccae8)
![swap_used_(mb)_plot](https://github.com/user-attachments/assets/7575198f-3044-4ee3-af2e-5611744fa8c8)
</details>

<details>
<summary>SCA logs</summary>

````console
# tail /Library/Ossec/logs/ossec.log
2024/08/01 22:25:13 sca: INFO: Evaluation finished for policy '/Library/Ossec/ruleset/sca/cis_apple_macOS_13.x.yml'
2024/08/01 22:25:13 sca: INFO: Security Configuration Assessment scan finished. Duration: 6 seconds.
2024/08/01 22:30:07 sca: INFO: Starting Security Configuration Assessment scan.
2024/08/01 22:30:07 sca: INFO: Starting evaluation of policy: '/Library/Ossec/ruleset/sca/cis_apple_macOS_13.x.yml'
2024/08/01 22:30:13 sca: INFO: Evaluation finished for policy '/Library/Ossec/ruleset/sca/cis_apple_macOS_13.x.yml'
2024/08/01 22:30:13 sca: INFO: Security Configuration Assessment scan finished. Duration: 6 seconds.
2024/08/01 22:35:07 sca: INFO: Starting Security Configuration Assessment scan.
2024/08/01 22:35:07 sca: INFO: Starting evaluation of policy: '/Library/Ossec/ruleset/sca/cis_apple_macOS_13.x.yml'
2024/08/01 22:35:14 sca: INFO: Evaluation finished for policy '/Library/Ossec/ruleset/sca/cis_apple_macOS_13.x.yml'
2024/08/01 22:35:14 sca: INFO: Security Configuration Assessment scan finished. Duration: 7 seconds.
````

</details>

[metrics.zip](https://github.com/user-attachments/files/16465938/metrics.zip)


## Analysis

  | Benchmark   (macOS 13 Old) | New SCA (macOS   13 rework) | Difference
-- | -- | -- | --
CPU |   |   |  
`wazuh-modulesd` average   CPU percent | 1.2 | 2 | 0.8
Overall   average CPU percent | 30.1 | 13.83 | -16.27
Overall peak   CPU percent | 59.9 | 29.4 | -30.5
Memory |   |   |  
`wazuh   modulesd` average memory (MB) | 0 | 0 | 0
Overall   average memory used (MB) | 1947.44 | 2042.12 | 94.68
Overall peak   memory used peak (MB) | 1969.49 | 2043.32 | 73.83
Disk |   |   |  
Overall   average disk used (MB) | 15152.34 | 15152.34 | 0
Overall peak   disk used (MB) | 15152.34 | 15152.34 | 0
Overall   average disk read bytes | 3479.16 | 3566.08 | 86.92
Overall peak   disk read bytes | 3479.26 | 3566.08 | 86.82
Overall   average disk write bytes | 6686.76 | 7459.13 | 772.37
Overall peak   disk write bytes | 6688.5 | 7459.41 | 770.91

**Comment by IsExec** [comment]
# Issue Update

Updated the SCA by setting check 1.1 as "Not Implemented" - https://github.com/wazuh/wazuh/pull/24992/commits/4594177a311fcb8ded2d615269627e800b058208

**Comment by ooniagbi** [comment]
LGTM!

**Comment by ooniagbi** [comment]
Merged to 4.10.0
